src/serverSetup.ts

```diff
@@ -53,8 +53,11 @@ export async function installCodeServer(conn: SSHConnection, serverDownloadUrlTe
         serverDownloadUrlTemplate,
     });
 
+    const installDir = `$HOME/${vscodeServerConfig.serverDataFolderName}/install`;
+    const installScript = `${installDir}/${vscodeServerConfig.commit}.sh`
+
     logger.trace('Server install command:', installServerScript);
-    const commandOutput = await conn.exec(`bash << 'EOF'\n${installServerScript}\nEOF`);
+    const commandOutput = await conn.exec(`mkdir -p ${installDir} && echo '\n${installServerScript.replace(/'/g, '\'"\'"\'')}\n' >${installScript} && bash ${installScript}`);
```

Comment:
I'm curious why the replace `replace(/'/g, '\'"\'"\'')` is needed?

Reply:
No, it's because I'm using `echo` with single quote to avoid the variables to be replaced (if double quotes are used).
So I have to escape the single quote `'` with `'"'"'` (which is also escaped with `\` for Javascript).

Reply:
Thanks, I added some :lipstick: